source/sessions/driver-sessions.rst

```diff
@@ -1197,6 +1212,25 @@ Instead, we require users to pass session as a parameter to each function::
   collection.insertOne(session, {})
   session.endSession()
 
+Why does a network error cause the ``ServerSession`` to be discarded from the pool?
+-----------------------------------------------------------------------------------
+
+When a network error is encountered when executing an operation with a
+``ClientSession``, the operation may be left running on the server. Re-using
+this ``ServerSession`` can lead to parallel operations which violates the
+rule that a session must be used sequentially. This results in multiple
+problems:
+
+#. killSessions to end an earlier operation would surprisingly also end a
+   later operation.
+#. An otherwise unrelated operation that just happens to use that same server
+   session will potentially block waiting for the previous operation to
+   complete. For example, a transactional write will block a subsequent
+   transactional write.
+
+TODO: Justify why we allow applications to continue using an explicit session
```

Comment:
Do you plan to do this before closing the PR?  A bit weird to leave a TODO in a spec.